From 7d7c45e9be36ebda96ebf9eb282e835d5ac415ac Mon Sep 17 00:00:00 2001
From: Hunt Xu <mhuntxu@gmail.com>
Date: Tue, 5 Sep 2017 17:30:45 +0800
Subject: [PATCH 109/112] Sync #119 from devel to testing

PPTP: fix vpnservices checking for PPTP credentials

Fixes: redmine #10813

Signed-off-by: Hunt Xu <mhuntxu@gmail.com>
(cherry picked from commit ddd72d4f1addc72ffb2582862fb1eeb07c6a2395)
Signed-off-by: Hunt Xu <mhuntxu@gmail.com>
---
 neutron/services/vpn/plugin.py | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/neutron/services/vpn/plugin.py b/neutron/services/vpn/plugin.py
index 3a4cb5642..c32554d64 100644
--- a/neutron/services/vpn/plugin.py
+++ b/neutron/services/vpn/plugin.py
@@ -209,14 +209,16 @@ class VPNDriverPlugin(VPNPlugin, vpn_db.VPNPluginRpcDbMixin):
         return ipsec_site_connection
 
     def create_pptp_credential(self, context, pptp_credential):
-        self._check_drivers_for_pptp_credential(context, pptp_credential)
+        self._check_drivers_for_pptp_credential(
+            context, pptp_credential['pptp_credential'])
         return super(
             VPNDriverPlugin, self
         ).create_pptp_credential(context, pptp_credential)
 
     def update_pptp_credential(self, context, pptp_credential_id,
                                pptp_credential):
-        self._check_drivers_for_pptp_credential(context, pptp_credential)
+        self._check_drivers_for_pptp_credential(
+            context, pptp_credential['pptp_credential'])
         return super(
             VPNDriverPlugin, self
         ).update_pptp_credential(context, pptp_credential_id, pptp_credential)
-- 
2.11.0 (Apple Git-81)

